tests/cephfs/cephfs_vol_management/cephfs_vol_mgmt_subvol_idempotence_earmark.py

```diff
@@ -0,0 +1,279 @@
+import random
+import traceback
+
+from ceph.parallel import parallel
+from tests.cephfs.cephfs_utilsV1 import FsUtils
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    CephFS Subvolume Test
+
+    This test suite validates the functionality of the `ceph fs subvolume` command,
+    covering various scenarios including:
+
+    * Idempotence with and without parameters (modes, uid, gid)
+    * Creating subvolumes with isolated namespace
+    * Failure handling and cleanup of incomplete subvolumes
+    * Creating subvolumes with desired uid, gid, modes, with and without groups
+    * Creating subvolumes with invalid size (negative and 0)
+    * Creating subvolumes with --group_name=_nogroup
+    * Setting and getting earmarks during subvolume creation and on existing subvolumes
+    * Setting earmarks with group_name
+    * Setting and removing earmarks while IO is in progress
+    """
+    try:
+        tc = "CEPH-83604184"
+        log.info(f"Running CephFS tests for - {tc}")
+        fs_util = FsUtils(ceph_cluster)
+        clients = ceph_cluster.get_ceph_objects("client")
+        config = kw.get("config")
+        fs_util.auth_list(clients)
+        build = config.get("build", config.get("rhbuild"))
+        fs_util.prepare_clients(clients, build)
+        client1 = clients[0]
+        vol_name = f"cephfs_subvolume_test_{random.randint(0, 1000)}"
+        fs_util.create_fs(client1, vol_name)
+        random_str = "".join(random.choices("abcdefghijklmnopqrstuvwxyz", k=5))
+        fuse_mounting_dir_1 = f"/mnt/cephfs_fuse_{random_str}"
+        fs_util.fuse_mount(
+            [client1], fuse_mounting_dir_1, extra_params=f" --client_fs {vol_name}"
+        )
+        subvol_name = f"subvol1_{random_str}"
+        # Idempotence default - same command 2-3 times
+        for _ in range(3):
+            command = f"ceph fs subvolume create {vol_name} {subvol_name}"
```

Comment:
for subvolume create we have existing lib(create_subvolume) which has below options

- vol_name:
- subvol_name:                 
- size : str in mb
- group_name : str
- pool_layout : str
- uid : str
- gid : str
- mode : str
- namespace_isolated : boolean
- check_ec = True. 

can we use the same, since this has been called multiple times in the script?

Reply:
yes we can do that as well